presto-orc/src/test/java/com/facebook/presto/orc/TestMapFlatSelectiveStreamReader.java

```diff
@@ -0,0 +1,641 @@
+                INTEGER,
+                ExpectedValuesBuilder.get(Function.identity()).setNullRowsFrequency(ALL));
+    }
+
+    // Some maps are empty
+    @Test
+    public void testWithEmptyMaps()
+            throws Exception
+    {
+        runTest(
+                "test_flat_map/flat_map_some_empty_maps.dwrf",
+                INTEGER,
+                ExpectedValuesBuilder.get(Function.identity()).setEmptyMapsFrequency(SOME));
+    }
+
+    // All maps are empty
+    @Test
+    public void testWithAllEmptyMaps()
+            throws Exception
+    {
+        runTest(
+                "test_flat_map/flat_map_all_empty_maps.dwrf",
+                INTEGER,
+                ExpectedValuesBuilder.get(Function.identity()).setEmptyMapsFrequency(ALL));
+    }
+
+    @Test
+    public void testMixedEncodings()
+            throws Exception
+    {
+        // Values associated with one key are direct encoded, and all other keys are
+        // dictionary encoded.  The dictionary encoded values have occasional values that only appear once
+        // to ensure the IN_DICTIONARY stream is present, which means the checkpoints for dictionary encoded
+        // values will have a different number of positions compared to direct encoded values.
+        runTest("test_flat_map/flat_map_mixed_encodings.dwrf",
+                INTEGER,
+                ExpectedValuesBuilder.get(Function.identity()).setMixedEncodings());
+    }
+
+    @Test
+    public void testIntegerWithMissingSequences()
+            throws Exception
+    {
+        // The additional sequences IDs for a flat map aren't a consecutive range [1,N], the odd
+        // sequence IDs have been removed.  This is to simulate the case where a file has been modified to delete
+        // certain keys from the map by dropping the ColumnEncodings and the associated data.
+        runTest("test_flat_map/flat_map_int_missing_sequences.dwrf",
+                INTEGER,
+                ExpectedValuesBuilder.get(Function.identity()).setMissingSequences());
+    }
+
+    private <K, V> void runTest(String testOrcFileName, Type type, ExpectedValuesBuilder<K, V> expectedValuesBuilder)
+            throws Exception
+    {
+        runTest(testOrcFileName, type, type, expectedValuesBuilder);
+    }
+
+    private <K, V> void runTest(String testOrcFileName, Type keyType, Type valueType, ExpectedValuesBuilder<K, V> expectedValuesBuilder)
+            throws Exception
+    {
+        List<Map<K, V>> expectedValues = expectedValuesBuilder.build();
+
+        Type mapType = mapType(keyType, valueType);
+
+        OrcPredicate orcPredicate = createOrcPredicate(0, mapType, expectedValues, OrcTester.Format.DWRF, true);
+
+        runTest(testOrcFileName, mapType, expectedValues, orcPredicate, Optional.empty(), ImmutableList.of());
+        runTest(testOrcFileName, mapType, expectedValues.stream().filter(map -> map == null).collect(toList()), orcPredicate, Optional.of(IS_NULL), ImmutableList.of());
```

Comment:
nit: `map -> map == null` -> `Objects::isNull` (or `Objects::nonNull`). Same for other places